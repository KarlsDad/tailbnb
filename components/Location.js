export default ({ location, price }) => {
  return (
    <div className='py-3 my-2'>
      <h3 className='font-semibold'>{location}</h3>
      <p className='font-normal text-gray-650 text-sm'>
        ${price}/night average
      </p>
    </div>
  );
};
