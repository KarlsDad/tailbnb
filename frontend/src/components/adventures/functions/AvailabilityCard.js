import React from 'react';

export default () => {
  return (
    <div className='w-full flex justify-start items-start flex-wrap'>
      <div className='w-23p bg-white shadow-md rounded-xl mr-4'>
        <div className='m-8'>
          <h3
            style={{ fontFamily: 'airbnb-medium' }}
            className='text-gray-750 text-2xl py-1'>
            Mar 03
          </h3>
          <p
            style={{ fontFamily: 'airbnb-book' }}
            className='text-gray-750 text-lg'>
            Tuesday
          </p>
          <p
            style={{ fontFamily: 'airbnb-book' }}
            className='text-gray-750 text-lg'>
            From $130/person
          </p>
          <button
            style={{ fontFamily: 'airbnb-bold' }}
            className='text-green-850 mt-8'>
            Book
          </button>
        </div>
      </div>
    </div>
  );
};
