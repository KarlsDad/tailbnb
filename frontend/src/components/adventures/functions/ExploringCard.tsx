import * as React from 'react';

interface Props {
  title: string;
  experienceNumber: number;
}

export const ExploringCard: React.FC<Props> = ({ title, experienceNumber }) => {
  return (
    <div className='bg-white shadow-sm rounded mr-4 border border-gray-300 mt-4'>
      <div className='px-5 py-5 mr-5'>
        <h3 style={{ fontFamily: 'airbnb-bold' }} className='text-gray-750'>
          {title}
        </h3>
        <p
          style={{ fontFamily: 'airbnb-medium' }}
          className='text-xs text-gray-650'>
          {experienceNumber} experiences
        </p>
      </div>
    </div>
  );
};
