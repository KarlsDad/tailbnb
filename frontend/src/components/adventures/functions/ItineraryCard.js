import React from 'react';

// Images
const detail2 = require('../../../../public/img/high/staydetail2.jpg');

export default () => {
  return (
    <div
      style={{ fontFamily: 'airbnb-book' }}
      className='px-5 text-gray-750 border-l border-gray-800 relative z-10'>
      <div style={{ marginBottom: 6 }}>
        <div
          style={{
            height: 7,
            width: 7,
            top: -7,
            left: -3.6
          }}
          className='absolute z-50'>
          <svg
            className='h-full w-full'
            xmlns='http://www.w3.org/2000/svg'
            viewBox='0 0 512 512'>
            <g>
              <g>
                <path
                  d='M256,0C115.03,0,0,115.05,0,256c0,140.97,115.05,256,256,256c140.97,0,256-115.05,256-256C512,115.03,396.95,0,256,0z
			 M256,482C131.383,482,30,380.617,30,256S131.383,30,256,30s226,101.383,226,226S380.617,482,256,482z'
                />
              </g>
            </g>
          </svg>
        </div>

        <p className='mb-2'>Day 1</p>
        <div className='w-full md:flex justify-start items-start pb-8'>
          <div className='md:w-5/12'>
            <img className='h-full w-full' src={detail2} alt='' />
          </div>

          <div className='w-full md:w-7/12 md:pl-5'>
            <p
              style={{ fontFamily: 'airbnb-bold' }}
              className='pt-3 pb-2 md:pt-0 md:pb-0 text-base'>
              Marrakech-Oughla - Tinzerte
            </p>
            <p className='text-base'>
              Marrakech to imi oughlad small village situated 1,450m at the base
              of Toubkal. Here we begin the gentle accent.
            </p>
            <button
              style={{ fontFamily: 'airbnb-bold' }}
              className='text-green-850 text-sm mt-4'>
              Show more
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

{
  /* <div
        style={{ height: 7, width: 7, bottom: -7, left: -3.6 }}
        className='absolute z-100'>
        <svg
          className='h-full w-full'
          xmlns='http://www.w3.org/2000/svg'
          viewBox='0 0 512 512'>
          <g>
            <g>
              <path
                d='M256,0C115.03,0,0,115.05,0,256c0,140.97,115.05,256,256,256c140.97,0,256-115.05,256-256C512,115.03,396.95,0,256,0z
			 M256,482C131.383,482,30,380.617,30,256S131.383,30,256,30s226,101.383,226,226S380.617,482,256,482z'
              />
            </g>
          </g>
        </svg>
      </div> */
}
