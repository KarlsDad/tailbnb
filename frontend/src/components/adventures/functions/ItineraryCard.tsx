import * as React from 'react';

// Images
const detail2 = require('../../../../public/img/high/staydetail2.jpg');

interface Props {
  is_last: boolean;
  day: number;
  title: string;
  description: string;
}

export const ItineraryCard: React.FC<Props> = ({
  is_last,
  day,
  title,
  description
}) => {
  return (
    <>
      {is_last ? (
        <div
          style={{ fontFamily: 'airbnb-book' }}
          className='px-5 text-gray-750 relative'>
          <div style={{ marginBottom: 6 }}>
            <div
              style={{
                height: 7,
                width: 7,
                top: -6.5,
                left: -3.4
              }}
              className='absolute z-50'>
              <svg
                className='h-full w-full'
                xmlns='http://www.w3.org/2000/svg'
                viewBox='0 0 512 512'>
                <g>
                  <g>
                    <path
                      d='M256,0C115.03,0,0,115.05,0,256c0,140.97,115.05,256,256,256c140.97,0,256-115.05,256-256C512,115.03,396.95,0,256,0z
			 M256,482C131.383,482,30,380.617,30,256S131.383,30,256,30s226,101.383,226,226S380.617,482,256,482z'
                    />
                  </g>
                </g>
              </svg>
            </div>

            <p className='mb-2'>Day {day}</p>
            <div className='w-full md:flex justify-start items-start pb-8'>
              <div className='md:w-5/12'>
                <img className='h-full w-full' src={detail2} alt='' />
              </div>

              <div className='w-full md:w-7/12 md:pl-5'>
                <p
                  style={{ fontFamily: 'airbnb-bold' }}
                  className='pt-3 pb-2 md:pt-0 md:pb-0 text-base'>
                  {title}
                </p>
                <p className='text-base'>{description}</p>
                <button
                  style={{ fontFamily: 'airbnb-bold' }}
                  className='text-green-850 text-sm mt-4'>
                  Show more
                </button>
              </div>
            </div>
          </div>
        </div>
      ) : (
        <div
          style={{ fontFamily: 'airbnb-book' }}
          className='px-5 text-gray-750 border-l border-gray-800 relative'>
          <div style={{ marginBottom: 6 }}>
            <div
              style={{
                height: 7,
                width: 7,
                top: -7,
                left: -3.6
              }}
              className='absolute'>
              <svg
                className='h-full w-full'
                xmlns='http://www.w3.org/2000/svg'
                viewBox='0 0 512 512'>
                <g>
                  <g>
                    <path
                      d='M256,0C115.03,0,0,115.05,0,256c0,140.97,115.05,256,256,256c140.97,0,256-115.05,256-256C512,115.03,396.95,0,256,0z
			 M256,482C131.383,482,30,380.617,30,256S131.383,30,256,30s226,101.383,226,226S380.617,482,256,482z'
                    />
                  </g>
                </g>
              </svg>
            </div>

            <p className='mb-2'>Day {day}</p>
            <div className='w-full md:flex justify-start items-start pb-8'>
              <div className='md:w-5/12'>
                <img className='h-full w-full' src={detail2} alt='' />
              </div>

              <div className='w-full md:w-7/12 md:pl-5'>
                <p
                  style={{ fontFamily: 'airbnb-bold' }}
                  className='pt-3 pb-2 md:pt-0 md:pb-0 text-base'>
                  {title}
                </p>
                <p className='text-base'>{description}</p>
                <button
                  style={{ fontFamily: 'airbnb-bold' }}
                  className='text-green-850 text-sm mt-4'>
                  Show more
                </button>
              </div>
            </div>
          </div>
        </div>
      )}
    </>
  );
};
