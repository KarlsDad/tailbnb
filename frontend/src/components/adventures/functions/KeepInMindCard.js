import React from 'react';

export default ({ title, description, order }) => {
  return (
    <>
      {order === 'top' ? (
        <div className='w-full flex pb-4 border-b border-gray-300'>
          <div className='w-1/3'>
            <h3 style={{ fontFamily: 'airbnb-medium' }} className='text-2xl'>
              {title}
            </h3>
          </div>
          <div className='w-2/3'>
            <p style={{ fontFamily: 'airbnb-book' }}>{description}</p>
            <button
              style={{ fontFamily: 'airbnb-medium' }}
              className='text-green-850 text-sm my-3'>
              Learn more
            </button>
          </div>
        </div>
      ) : (
        <>
          {order === 'bottom' ? (
            <div className='w-full flex pb-4 pt-6'>
              <div className='w-1/3'>
                <h3
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='text-2xl'>
                  {title}
                </h3>
              </div>
              <div className='w-2/3'>
                <p style={{ fontFamily: 'airbnb-book' }}>{description}</p>
                <button
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='text-green-850 text-sm my-3'>
                  Learn more
                </button>
              </div>
            </div>
          ) : (
            <div className='w-full flex pb-4 pt-6 border-b border-gray-300'>
              <div className='w-1/3'>
                <h3
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='text-2xl'>
                  {title}
                </h3>
              </div>
              <div className='w-2/3'>
                <p style={{ fontFamily: 'airbnb-book' }}>{description}</p>
                <button
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='text-green-850 text-sm my-3'>
                  Learn more
                </button>
              </div>
            </div>
          )}
        </>
      )}
    </>
  );
};
