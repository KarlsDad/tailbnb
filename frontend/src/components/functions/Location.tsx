import * as React from 'react';

interface Prop {
  location: string;
  price: number;
}

export const Location: React.FC<Prop> = ({ location, price }) => {
  return (
    <div className='w-90p py-3 my-2'>
      <div>
        <h3 style={{ fontFamily: 'airbnb-medium' }} className='text-gray-750'>
          {location}
        </h3>
      </div>
      <div style={{ fontFamily: 'airbnb-book' }} className=''>
        <p className='text-xs text-gray-650 md:text-sm whitespace-normal'>
          ${price}/night average
        </p>
      </div>
    </div>
  );
};
