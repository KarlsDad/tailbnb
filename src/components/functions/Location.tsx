import * as React from 'react';

interface Prop {
  location: string;
  price: number;
}

export const Location: React.FC<Prop> = ({ location, price }) => {
  return (
    <div className='py-3 my-2'>
      <div>
        <h3
          style={{ fontFamily: 'airbnb-medium' }}
          className='text-gray-850 text-lg'>
          {location}
        </h3>
      </div>
      <div style={{ fontFamily: 'airbnb-book' }} className=''>
        <p className='text-xs text-gray-650 md:text-sm whitespace-normal'>
          ${price}/night average
        </p>
      </div>
    </div>
  );
};
