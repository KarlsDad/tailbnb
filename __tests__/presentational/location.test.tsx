import * as React from 'react';
import { shallow } from 'enzyme';

// Component
import { Location } from '../../src/components/functions/Location';

describe('Location', () => {
  it('should render correct in "debug" mode', () => {
    const info = {
      location: 'testlocation',
      price: 10
    };
    const component = shallow(<Location {...info} />);
    expect(component).toMatchSnapshot();
  });
});
