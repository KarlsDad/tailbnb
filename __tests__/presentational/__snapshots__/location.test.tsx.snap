// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location should render correct in "debug" mode 1`] = `ShallowWrapper {}`;
